tests/chainerx_tests/unit_tests/routines_tests/test_linalg.py

```diff
@@ -93,3 +93,35 @@ def test_dot_invalid(is_module, xp, device, a_shape, b_shape, dtype):
         return xp.dot(a, b)
     else:
         return a.dot(b)
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    # Special shapes
+    chainer.testing.product({
+        'a_size': [2, ],
```

Comment:
Could you cover edge cases such as an empty array? And to do so, how about parameterize over the complete shape instead of only specifying a dimension? Also, could you cover invalid inputs to test that correct errors are raised?

Reply:
Empty and non-square arrays are added. Errors are raised if a matrix is not positive definite and non-square.

Reply:
I think it's better to separate tests that expect failures? It's mostly due to the way the test helper classes are implemented but by registering certain failures as allowed, configurations that shouldn't raise these error might do so while still passing the tests.

Reply:
You are right, it is safer to have them separate.